### PR TITLE
Migrate FF4J DB

### PR DESCRIPTION
**Pull request by doncem**
Enable migrations:

- terraform code for db
- flyway no-op strategy
- datasource configuration
- ddl from ff5j library

bonus:

- fix test package name

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -8,6 +12,28 @@ resource "azurerm_resource_group" "rg" {
...
+  s2s_url                = "http://rpe-service-auth-provider-${local.local_env}.service.${local.local_ase}.internal"
+
+  previewVaultName       = "${var.product}-${var.component}"
+  nonPreviewVaultName    = "${var.product}-${var.component}-${var.env}"
+  vaultName              = "${(var.env == "preview" || var.env == "spreview") ? local.previewVaultName : local.nonPreviewVaultName}"
+
+  db_connection_options  = "?ssl=true"
+}
+
+module "db" {
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: let's  name this variable something like `feature-toggle-db`

File: src/main/java/uk/gov/hmcts/reform/feature/exception/PendingMigrationScriptException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.feature.exception;
+
+public class PendingMigrationScriptException extends RuntimeException {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/feature/exception/PendingMigrationScriptException.java]: Do we need to extend UnknowErrorCodeException? Else do we get exceptions in app insights ?

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/feature/exception/PendingMigrationScriptException.java]: I've [disabled](https://github.com/hmcts/feature-toggle-api/blob/master/infrastructure/main.tf#L23-L24) those 😆

File: src/main/resources/db/migration/V1__FF4J_schema_from_library.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
+-- Main Table to store Features
```
Comment (rkondratowicz) [inline src/main/resources/db/migration/V1__FF4J_schema_from_library.sql]: add leading zeros in version in file name

Comment (doncem) [reply src/main/resources/db/migration/V1__FF4J_schema_from_library.sql]: oh, sillies mes
> I meant the migration file name

**Pull request merged**